### DISCUSSION TITLE
Support for HmIP-WGC

### DISCUSSION THREAD

**Discussion by regnets** [SukramJ/aiohomematic#1002] [Ideas]
I would like to request support for the HmIP-WGC Garage Door Button.

If needed, i can provide further details/log for that device. If a developer has a garage and needs this device i also would purchase this device as an incentive.

**Comment by SukramJ** [comment]
What is missing?
If you want to act on keypress events then you have to use device triggers in automation, as for all other push button devices.
More information can be found in the documentation:
https://github.com/danielperna84/custom_homematic#devices-with-buttons
https://github.com/danielperna84/custom_homematic#events-for-homematicip-devices

**Reply by regnets** [reply]
Wow, maybe i was to blind to find the device. I still was under the impression that this device isn't show up on HA. I am really sorry. I didn't find it for the last few days and i searched arround to find why this device wasn't supported yet. I'll close that issue.

**Reply by UkeHa** [reply]
@SukramJ, i'm trying to add a button in home assistant that will toggle the garage door. I fail to see how i can get the addon to do that. If i see things correctly i should see channel 3 on the HmIP-WGC and toggle it. I don't see that channel in homeassistant. 

Another way would be to trigger an automation/service like this, right?

```
service: homematicip_local.set_device_value
data:
  device_id: ???
  parameter: PRESS_LONG
  value: "true"
  value_type: boolean
  channel: 3
```

what would be the device_id in this case? And how do i get a button that opens the door?

**Reply by UkeHa** [reply]
@regnets, did you figure it out? If so, could you provide some insight?
